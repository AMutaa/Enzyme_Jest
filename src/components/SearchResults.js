import React from 'react';
import PropTypes from 'prop-types';

const SearchResults = ({ articles = [] }) => (
  <ul>
    {articles.map(({ webUrl, webTitle }) => (
      <li>
        <a href={webUrl} target='_blank' rel='noopener noreferrer'>
          {webTitle}
        </a>
      </li>
    ))}
  </ul>
)

SearchResults.PropTypes = {
  articles: PropTypes.array
}

export default SearchResults