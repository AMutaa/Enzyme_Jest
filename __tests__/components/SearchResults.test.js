import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import SearchResults from '../../src/components/SearchResults';

const mockArticles = [
  {
    url: 'http://google.com',
    title: 'This is google'
  }
]

//Snapshot testing
describe('SearchResults component', () => {
  test('renders', () => {
    const wrapper = shallow(<SearchResults articles={mockArticles} />)
    expect(wrapper).toMatchSnapshot();
  })
})

//empty array of articles, component still renders
test('returns the default empty array when there is no data to map through', () => {
  const wrapper = shallow(<SearchResults />);
  expect(wrapper).toMatchSnapshot();
})

//no break without articles
test('does not break without articles', () => {
  const wrapper = shallow(<SearchResults />);
  expect(wrapper.find('li')).toHaveLength(0);
})

test('does not break with an empty array', () => {
  const wrapper = shallow(<SearchResults articles={[]} />);
  expect(wrapper.find('li')).toHaveLength(0);
})
