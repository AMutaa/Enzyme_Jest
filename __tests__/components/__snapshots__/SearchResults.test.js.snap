// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResults
    articles={
      Array [
        Object {
          "title": "This is google",
          "url": "http://google.com",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <a
            href="http://google.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            This is google
          </a>
        </li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "This is google",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="http://google.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            This is google
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is google",
            "href": "http://google.com",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "This is google",
          "type": "a",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <a
              href="http://google.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              This is google
            </a>
          </li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "This is google",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="http://google.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              This is google
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "This is google",
              "href": "http://google.com",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "This is google",
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`returns the default empty array when there is no data to map through 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResults />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
